ar_track_alvar/CMakeLists.txt

```diff
@@ -1,183 +1,187 @@
+find_package(perception_pcl  REQUIRED)
+find_package(pcl_ros)
+find_package(pcl_conversions REQUIRED)
+find_package(ar_track_alvar_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(rosbag2_bag_v2_plugins REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(PCL REQUIRED QUIET COMPONENTS common io)
+find_package(Eigen3 REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
+
+include_directories(include)
+
+# Kinect filtering code
+set(KINECT_FILTERING_TARGETS kinect_filtering medianFilter)
+set(ALVAR_TARGETS ar_track_alvar individualMarkers individualMarkersNoKinect trainMarkerBundle findMarkerBundles findMarkerBundlesNoKinect createMarker)
+
+set(dependencies
+  OpenCV
+  tf2_ros 
+  tf2
+  pcl_ros
+  pcl_conversions 
+  std_msgs 
+  tinyxml_vendor
+  image_transport
+  perception_pcl 
+  visualization_msgs 
+  rclcpp 
+  resource_retriever 
+  geometry_msgs 
+  tf2_geometry_msgs
+  cv_bridge 
+  sensor_msgs
+  ar_track_alvar_msgs
 )
 
-include_directories(include 
-                    ${catkin_INCLUDE_DIRS}
+include_directories(include
                     ${OpenCV_INCLUDE_DIRS}
                     ${TinyXML_INCLUDE_DIRS}
-
+                    ${PCL_COMMON_INCLUDE_DIRS}
 )
 
-set(GENCPP_DEPS ar_track_alvar_msgs_gencpp std_msgs_gencpp sensor_msgs_gencpp geometry_msgs_gencpp visualization_msgs_gencpp)
-
-add_library(ar_track_alvar
-    src/Camera.cpp
-    src/CaptureDevice.cpp
-    src/Pose.cpp
-    src/Marker.cpp
-    src/MarkerDetector.cpp
+add_library(${PROJECT_NAME}
     src/Bitset.cpp
-    src/Rotation.cpp
-    src/CvTestbed.cpp
+    src/Camera.cpp
     src/CaptureDevice.cpp
     src/CaptureFactory.cpp
     src/CaptureFactory_unix.cpp
-    src/FileFormatUtils.cpp
-    src/Threads.cpp
-    src/Threads_unix.cpp
-    src/Mutex.cpp
-    src/Mutex_unix.cpp
     src/ConnectedComponents.cpp
-    src/Line.cpp src/Plugin.cpp
-    src/Plugin_unix.cpp
+    src/CvTestbed.cpp
     src/DirectoryIterator.cpp
     src/DirectoryIterator_unix.cpp
     src/Draw.cpp
-    src/Util.cpp
+    src/FileFormatUtils.cpp
     src/Filter.cpp
     src/Kalman.cpp
-    src/kinect_filtering.cpp
-    src/Optimization.cpp
+    src/Line.cpp
+    src/Marker.cpp
+    src/MarkerDetector.cpp
     src/MultiMarker.cpp
     src/MultiMarkerBundle.cpp
-    src/MultiMarkerInitializer.cpp)
-target_link_libraries(ar_track_alvar ${OpenCV_LIBS} ${TinyXML_LIBRARIES} ${catkin_LIBRARIES})
-add_dependencies(ar_track_alvar ${GENCPP_DEPS})
-
-# Kinect filtering code
-set(KINECT_FILTERING_TARGETS kinect_filtering medianFilter)
+    src/MultiMarkerInitializer.cpp
+    src/Mutex.cpp
+    src/Mutex_unix.cpp
+    src/Optimization.cpp
+    src/Plugin.cpp
+    src/Plugin_unix.cpp
+    src/Pose.cpp
+    src/Rotation.cpp
+    src/Threads.cpp
+    src/Threads_unix.cpp
+    src/Util.cpp
+)
+    
+target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBRARIES} tinyxml)
```

Comment:
To make it compilable, I added  `find_package(PCL REQUIRED)` on top, and alse appended `${PCL_LIBRARIES}` here.

Reply:
Awesome thanks!